### ISSUE TITLE
[2.5] EKS upgrades loop without a default AMI list for v1.19 onwards

### ISSUE THREAD

**Issue by deniseschannon** [rancher/rancher#35725]
Backport https://github.com/rancher/rancher/issues/35295

SURE-3342

**Comment by Auston-Ivison-Suse** [comment]
**Setup For Reproduction**
- Rancher version: v2.5.11
- Rancher install: docker
- Downstream clusters: EKS
   - Kubernetes version: 1.18/1.19

**Setup For Validation**
- Rancher version: v2.5-head(07a745a)
- Rancher install: docker
- Downstream clusters: EKS 
   - Kubernetes version: 1.18/1.19

**Steps For Reproduction**

1. Go to  https://rancherserver/g/clusters/add/launch/amazoneks
2. select regions as well as kubernetes versions: us-east2 and k8s: v1.18/v1.19 (One cluster for each version of k8s)
3. Proceed with all other defaults.
4. Create cluster
5. wait for error to show up which will be listed as follows: (this will happen when provisioning the v1.19 cluster and will happening when upgrading the v1.18 cluster to v1.18) 

```
Default ami of region us-east-2 for kubernetes version 1.19 is not set
````

**Steps For Validation**

1. Go to  https://rancherserver/g/clusters/add/launch/amazoneks
2. select regions as well as kubernetes versions: us-east2 and k8s: v1.18/v1.19 (One cluster for each version of k8s)
3. Proceed with all other defaults.
4. Create cluster
5. The v1.19 cluster will become active with no issues and once you upgrade the v1.18 cluster to v1.19 it should also become active with no issues. 

**Results**
The clusters provision correctly and have a default ami set in aws for the v1.19 kubernetes version

**Notes**
It should be noted that https://rancherserver/g/clusters/add/launch/amazoneksv2 allows for clusters to be provisioned in v2.5.11.

**Comment by throrin19** [comment]
Same error in eu-west-1 region

**Comment by thedadams** [comment]
@throrin19 This change was included in v2.5.12, which was just released.

**Comment by throrin19** [comment]
I see that, But How remove the error and correctly launch the update of my cluster ? I try to reedit the cluster and validate but nothing happen
